plugins/vpc-eni/network/hnswrapper/hcsshim_windows.go

```diff
@@ -0,0 +1,242 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package hnswrapper
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/Microsoft/hcsshim"
+	log "github.com/cihub/seelog"
+)
+
+type builder struct{}
+
+// NewWindowsNetworkBuilder returns a new instance of NetworkBuilder for Windows.
+func NewWindowsNetworkBuilder() WindowsNetworkBuilder {
+	return &builder{}
```

Comment:
I'm trying to understand why we need so many indirection abstraction layers. Plugin has a network.Builder, which is a NetBuilder, which is an hnswrapper.WindowsNetworkBuilder, which is a builder. Can you please help me understand what purpose these indirection layers serve? Why doesn't Plugin create directly a builder?